go/indexer/db/db.go

```diff
@@ -772,19 +557,41 @@ func (d *Database) GetWithdrawalsByAddress(address l2common.Address, page Pagina
 		withdrawals[i].Batch = batch
 	}
 
-	return withdrawals, nil
-}
+	const selectWithdrawalCountStatement = `
+	SELECT
+		count(*)
+	FROM withdrawals
+		INNER JOIN l2_blocks ON withdrawals.block_hash=l2_blocks.hash
+		INNER JOIN l2_tokens ON withdrawals.l2_token=l2_tokens.address
+	WHERE withdrawals.from_address = $1;
+	`
 
-type L1BlockLocator struct {
-	Number uint64      `json:"number"`
-	Hash   common.Hash `json:"hash"`
-}
+	var count uint64
+
+	err = txn(d.db, func(tx *sql.Tx) error {
+		queryStmt, err := tx.Prepare(selectWithdrawalCountStatement)
```

Comment:
Separately - I don't think you need to do an explicit `Prepare` here, `tx.QueryRow(selectWithdrawalCountStatement)` will work fine. Prepared statements allow the DB engine to avoid re-parsing the query, which can improve performance somewhat but is a no-op here since the statement is re-prepared on every invocation of this method.

Reply:
Will do.